codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/protocol/ServerProtocolTestGenerator.kt

```diff
@@ -244,27 +241,35 @@ class ServerProtocolTestGenerator(
         writeInline("let output =")
         instantiator.render(this, expectedShape, testCase.params)
         write(";")
+        val operationName = if (expectedShape.hasTrait(ErrorTrait::class.java)) {
+            "${operationSymbol.name}${ServerHttpProtocolGenerator.OPERATION_ERROR_WRAPPER_SUFFIX}"
+        } else {
+            "${operationSymbol.name}${ServerHttpProtocolGenerator.OPERATION_OUTPUT_WRAPPER_SUFFIX}"
+        }
```

Comment:
I am not 100% sure I should use `expectedShape.hasTrait(ErrorTrait::class.java)` to understand if the shape is an error or not, but we don't have tests for error types as of now.

Reply:
This is fine. It's equivalent to

```
expectedShape.hasTrait<ErrorTrait>()
```

where you have to import

```
import software.amazon.smithy.rust.codegen.util.hasTrait
```

which is a more Kotlin-ish way of doing this and is what the rest of the codebase is using, but it actually desugars to what you wrote.

Reply:
Thanks, I'll use the more idiomatic `hasTrait<ErrorTrait>()`

I tried, but I guess I forgot the import :\